### PR TITLE
4750 database monitor & deceased contact title

### PR DESCRIPTION
**Pull request by latin-panda**
# Description

Migrated database connection monitor service and database closed modal.
Fixed deceased contact title in mobile.

https://github.com/medic/angular10-migration/issues/94
https://github.com/medic/angular10-migration/issues/104

# Code review items

- Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- Tested: Unit and/or e2e where appropriate
- Internationalised: All user facing text
- Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

**Comment by latin-panda** [comment]
Hi @njogz Can you please take a look at this? I made a little refactor in Contact Decease and Contact Effect that I think it'd be nice to know your feeback. 
Additionally it migrates the Database connection monitor, you can see how to test it in the comment [here](https://github.com/medic/angular10-migration/issues/94)
Thank you!!

File: webapp/src/css/inbox.less [hunk 1]
```diff
@@ -9,7 +9,6 @@
 @import 'font';
 @import 'mixins';
 @import 'theme';
-@import 'angular';
```
Comment (latin-panda) [inline webapp/src/css/inbox.less:12]: This is removed because we no longer have cloak in new angular.

File: webapp/src/ts/app.component.ts [hunk 2]
```diff
@@ -364,6 +368,18 @@ export class AppComponent implements OnInit {
...
 
+  private watchDatabaseConnection() {
+    this.databaseConnectionMonitorService
+      .listenForDatabaseClosed()
+      .subscribe(() => {
+        this.modalService
+          .show(DatabaseClosedComponent)
+          .catch(() => {});
+
+        // ToDo: closeDropdowns();
```
Comment (latin-panda) [inline webapp/src/ts/app.component.ts:379]: This closeDropdowns() hasn't been migrated yet. Part of the app component migration ticket.

File: webapp/src/ts/components/navigation/navigation.component.ts [hunk 3]
```diff
@@ -55,7 +55,6 @@ export class NavigationComponent implements OnInit, OnDestroy {
   navigateBack() {
     const routeSnapshot = this.routeSnapshotService.get();
     if (routeSnapshot.data.name === 'contacts.deceased') {
-      // todo check if this works when we have migrated the contacts tabs
```
Comment (latin-panda) [inline webapp/src/ts/components/navigation/navigation.component.ts:58]: I've checked, it's working.

File: webapp/src/ts/effects/contacts.effects.ts [hunk 4]
```diff
@@ -66,12 +68,18 @@ export class ContactsEffects {
...
           return []; // return an empty stream if there is no selected contact
         }
+
         const refreshing = previousSelectedContact?.doc?._id === selected.id;
         this.globalActions.settingSelected(refreshing);
         this.globalActions.clearCancelCallback();
-        const options = { getChildPlaces: true };
-        const title = (selected.type && selected.type.name_key) || 'contact.profile';
+
+        const routeSnapshot = this.routeSnapshotService.get();
```
Comment (latin-panda) [inline webapp/src/ts/effects/contacts.effects.ts:76]: My point of view:
The title shouldn't be set in effect & store, it's something that every page should be able to define freely. 
If in the future we add a new content component, 'contact new borns', then we need to add another condition here for the title to work properly, similar to what happened with 'contact decease'. Display components like 'navigation title' should ideally receive the data by input. Decreasing complexity in interaction between feature component, store, actions (unset and set title) and effects (usually complex).
Some other might be candidate for the same refactor (action bar?), something to think further and refactor in next iterations, I've added it in my to do list that I'll share after migration (fyi @craig-landry nice item consider later 🤩)

Comment (dianabarsan) [reply webapp/src/ts/effects/contacts.effects.ts:76]: I agree that the title should ideally be set in each component.

**Comment by latin-panda** [comment]
Hi @njogz can you please have a look? Wondering if it's all good to merge :)

**Review by njogz — APPROVED** [review]
Great job. Looks good.

**Pull request merged**